crashreports/crashreporter/CrashReportApp.cpp

```diff
@@ -0,0 +1,464 @@
+                {
+                    if (str[i] == 0)
+                        throw std::logic_error("unterminated string literal");
+                    else if (str[i] == '\"')
+                    {
+                        if (i > begin)
+                            result[key] = str.substr(begin, i - begin);
+                        else
+                            result[key] = std::string();
+                        ++i;
+                        state = TOKEN_VALUE;
+                        break;
+                    }
+                    ++i;
+                }
+            }
+            else if (str[i] == ',')
+            {
+                if (state != TOKEN_VALUE)
+                    throw std::logic_error("malformed parameters string: unexpected ',' symbol");
+                state = TOKEN_COMMA;
+                ++i;
+            }
+            else
+                throw std::logic_error("malformed parameters string");
+        }
+        if (state != TOKEN_VALUE)
+            throw std::logic_error("malformed parameters string");
+
+        return result;
+    }
+
+    void PrintMinidump(google_breakpad::Minidump& minidump)
+    {
+        google_breakpad::BasicSourceLineResolver resolver;
+        google_breakpad::MinidumpProcessor minidumpProcessor(nullptr, &resolver);
+        google_breakpad::MinidumpThreadList::set_max_threads(std::numeric_limits<uint32_t>::max());
+        google_breakpad::MinidumpMemoryList::set_max_regions(std::numeric_limits<uint32_t>::max());
+        
+        google_breakpad::ProcessState processState;
+        
+        if (minidumpProcessor.Process(&minidump, &processState) != google_breakpad::PROCESS_OK)
+        {
+            printf("Failed to process minidump");
+        }
+        else
+        {
+            google_breakpad::PrintProcessState(processState, true, &resolver);
+        }
+    }
+
+    wxString MakeDumpString(google_breakpad::Minidump& minidump, const wxString& temp)
+    {
+#if _WIN32
+        auto stream = _wfreopen(temp.wc_str(), L"w+", stdout);
+#else
+        auto stream = freopen(temp.utf8_str().data(), "w+", stdout);
+#endif
+        if (stream == NULL)
+            throw std::runtime_error("Failed to print minidump: cannot open temp file");
+        PrintMinidump(minidump);
+        fflush(stdout);
+
+        auto length = ftell(stream);
+        std::vector<char> bytes(length);
+        fseek(stream, 0, SEEK_SET);
+        fread(&bytes[0], 1, length, stream);
+        fclose(stream);
+        
+#if _WIN32
+        _wremove(temp.wc_str());
+#else
+        remove(temp.utf8_str().data());
+#endif
+
+        return wxString::From8BitData(&bytes[0], bytes.size());
+    }
+
+    wxString MakeHeaderString(google_breakpad::Minidump& minidump)
+    {
+        if (auto exception = minidump.GetException())
+        {
+            if (auto rawException = exception->exception())
+            {
+                // i18n-hint C++ programming assertion
```

Comment:
Comment should say "exception" not "assertion."  But the important thing is that there is a i18n-hint comment now, which will be in future updates of audacity.pot, and help the translator understand that the word means something in the technical sense of C++.